src/types.jl

```diff
@@ -35,15 +35,15 @@ abstract type Records end
 
 # Create a instance of a `Records` subtype, with all fields (assumed to be Vector)
 # containing `nrow` elements (initialised to undefined values, as they'll be overwritten).
-(::Type{R})(nrow) where {R <: Records} = R(map(T -> T(undef, nrow), fieldtypes(R))...)
+(::Type{R})(nrow) where {R <: Records} = R(map(T -> T(undef, (nrow,)), fieldtypes(R))...)
```

Comment:
todo: check why this change was needed (i think something to do with SentinelArrays but that needs checking)